### PR TITLE
DashboardScene: prevent panel hovel header crop with scenes

### LINKED ISSUE #84830: Panel menu not visible when panel has no title
The panel menu hides behind when:
* The panel is the first one in the dashboard, hides behind the fixed controls (variables, timepicker, etc.)
* The panel is below another panel (this also seems to be the case in the previous architecture)

I atatch a video with the issue. The first part is with the Scenes toggle off.

https://github.com/grafana/grafana/assets/74534993/09724af4-b47b-41c7-b53b-8345f79aecd1

### PR DESCRIPTION
**Pull request by Sergej-Vlasov**
- [x] adjusted `buildGridItemForPanel` to include `hoverHeaderOffset` for top panels
- [x] added `hoverHeaderOffsetBehavior` to adjust `hoverHeaderOffset` when moving panels

Relies on https://github.com/grafana/scenes/pull/674

Fixes #84830

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts [hunk 1]
```diff
@@ -475,6 +475,16 @@ export function buildGridItemForLibPanel(panel: PanelModel) {
   });
 }
 
+const hoverHeaderOffsetBehavior = (grid: DashboardGridItem) => {
+  grid.subscribeToState(() => {
```
Comment (dprokop) [inline public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:479]: you are creating a subscription here, that you never unsubscribe from. This needs to beccome :
```ts
const sub = grid.subscribeToState(() => {
....
```

and then your behavior needs to return a deactivation handler that will unsubscribe:

```ts
return () => {
  sub.unsubscribe();
}
```

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts [hunk 2]
```diff
@@ -475,6 +475,16 @@ export function buildGridItemForLibPanel(panel: PanelModel) {
...
+    grid.forEachChild((child) => {
+      if (child instanceof VizPanel && child.state.hoverHeader) {
+        child.setState({ hoverHeaderOffset: grid.state.y === 0 ? 0 : undefined });
```
Comment (dprokop) [inline public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:482]: Can we optimise this and only run `child.setState` if the `y` state of the grid item changed? `subscribeToState` receives new and previous state to perform comparison.

**Review by dprokop — CHANGES_REQUESTED** [review]
Looks good (and works well too), but some bits to improve.

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts [hunk 3]
```diff
@@ -475,6 +475,16 @@ export function buildGridItemForLibPanel(panel: PanelModel) {
   });
 }
 
+const hoverHeaderOffsetBehavior = (grid: DashboardGridItem) => {
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:478]: Can we extract this behavior to a different file? Just to remove noise from this one which is quite critical for Scene creation and it is frequently revisited :)

**Review by ivanortegaalba — COMMENTED** [review]
Looks good to me. I second Dominik's comments, but it looks good so far :)

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts (hunk 1 above)
Comment (Sergej-Vlasov) [reply public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts]: @dprokop  a bit weird but I don't see when  `sub.unsubscribe()` is fired. For example `console.log('unsubscribe')` never shows up. Did I miss something?
```typescript
return () => {
  console.log('unsubscribe');  
  sub.unsubscribe();
}
```

Comment (dprokop) [reply public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts]: it's going to be called when you remove a panel or navigate away from a dashboard or go to panel edit. Literally when scene object get's deactivated (https://grafana.com/developers/scenes/advanced-activation-handlers)

**Review by dprokop — APPROVED** [review]

**Pull request merged**